File: apps/plant_shop_python/models/order.py
# models/order.py

from dataclasses import dataclass
from datetime import datetime
from decimal import Decimal
from typing import List, Optional
from .order_item import OrderItem

"""
	Représente une commande dans la base de données.

	@param id Identifiant unique de la commande
	@param user_id Identifiant de l utilisateur ayant passé la commande
	@param total Montant total de la commande
	@param status Statut de la commande (pending, confirmed, shipped, delivered)
	@param created_at Date de création de la commande
	@param items Liste des articles de la commande (chargés séparément)
"""
@dataclass
class Order:
    id: int
    user_id: int
    total: Decimal
    status: str
    created_at: datetime
    items: Optional[List[OrderItem]] = None # Les items sont chargés séparément

File: apps/plant_shop_java/Structure_en_application_distribuée/plant_shop_java_distribuée/order-service/src/model/Order.java
import java.math.BigDecimal;
import java.time.Instant;

/**
 * Modèle représentant une commande.
 */
public final class Order {
    int id;
    int userId;
    BigDecimal total;
    String status;
    Instant createdAt;

    /**
	 * Constructeur complet.
	 * @param id Identifiant
	 * @param userId Identifiant de l'utilisateur
	 * @param total Montant total
	 * @param status Statut de la commande
	 * @param createdAt Date de création
	 */
	public Order(int id, int userId, BigDecimal total, String status, Instant createdAt) {
        this.id = id;
        this.userId = userId;
        this.total = total;
        this.status = status;
        this.createdAt = createdAt;
    }
}
